sdk/iot/hub/samples/README.md

```diff
@@ -0,0 +1,63 @@
+---
+page_type: sample
+languages:
+  - c
+products:
+  - azure
+  - azure-iot
+  - azure-iot-hub
+urlFragment: iot-hub-samples
+---
+
+# Azure IoT Hub Sample
+
+This document explains samples for the Azure Embedded C SDK IoT Hub Client and how to use them.
+
+## Key concepts
+
+Key concepts are explained in detail [here][SDK_README_KEY_CONCEPTS].
+
+## Samples for Azure IoT Hub Client APIs
+
+This document describes how to use samples and what is done in each sample.
+
+## Getting started
+
+Getting started explained in detail [here][SDK_README_GETTING_STARTED].
+
+## Examples
+
+The following section documents various examples. All of them use the [Eclipse Paho MQTT C client][Eclipse_Paho].
+
+1. [IoT Hub C2D][paho_c2d_sample]: Receive and view incoming C2D messages using the IoT Hub client.
```

Comment:
I went with alphabetical/the way the directory structure is set up. It's a toss up either way. Might let other clues in (I know it's a nit)?

Reply:
Basic SKUs don't support the other features; Telemetry is the most important feature (the rest may or may not be required by devices). Reading the doc, users should find the most important features at top.
I would argue that C2D be moved last: most people should use Methods and Twin instead of C2D.

Reply:
reordered